dhall/src/Dhall.hs

```diff
@@ -935,6 +1064,12 @@ instance Interpret a => Interpret [a] where
 instance Interpret a => Interpret (Vector a) where
     autoWith opts = vector (autoWith opts)
 
+instance (Interpret a, Ord a, Show a) => Interpret (Data.Set.Set a) where
```

Comment:
Can you document how this instance handles duplicates? Maybe just reference `setFromDistinctList`.

Reply:
There can't be any duplicates here because the source is a `Set` which turns into a Dhall `List`.

Reply:
I take that as an indication that we should find better names for `Interpret` and `Inject`. ;)

(Take a look at the definition on the line below)

Reply:
Oh my God, don't I feel silly!
Thank you for taking some of the blame away from me :)